### PR TITLE
Separate `Store` into smaller chunks

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

Currently, we have one monolithic trait `Store`. This PR splits the trait into 6

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1202538466812818/1202675758696579/f) _(internal)_

## 🔍 What does this change?

- Split `Store` into `DataTypeStore`, `PropertyTypeStore`, `LinkTypeStore`, `EntityTypeStore`, `EntityStore`, `LinkStore`
- Make `Store` an alias to the new stores
- Rename `DataBaseType` to `OnthologyDatabaseType`

Note, that this PR is not changing any logic, but due to the restructuring, the git-diff is quite big.

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/store/postgres/entity_type/mod.rs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+use tokio_postgres::GenericClient;
+
+use crate::{
+    ontology::AccountId,
+    store::{AsClient, EntityTypeStore, InsertionError, PostgresStore, UpdateError},
+    EntityType,
+};
+
+#[async_trait]
+impl<C: AsClient> EntityTypeStore for PostgresStore<C> {
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/entity_type/mod.rs:12]: No worries!
Yes, they would become a blanket implementation.

File: packages/graph/hash_graph/lib/graph/src/store/mod.rs [hunk 2]
```diff
@@ -174,8 +176,12 @@ impl fmt::Display for DatabaseConnectionInfo {
 ///
 /// In addition to the errors described in the methods of this trait, further errors might also be
 /// raised depending on the implementation, e.g. connection issues.
+pub trait Store =
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/mod.rs:179]: Actually, I'm not sure, if this is the reason or the fact that we return a concrete type from the pool by calling `acquire()`. It's most probably the latter.

File: packages/graph/hash_graph/lib/graph/src/store/postgres/onthology/read.rs [hunk 3]
```diff
@@ -0,0 +1,48 @@
+use async_trait::async_trait;
+use error_stack::{IntoReport, Result, ResultExt};
+use serde::Deserialize;
+use tokio_postgres::GenericClient;
+
+use crate::{
+    ontology::types::uri::VersionedUri,
+    store::{crud, postgres::onthology::OntologyDatabaseType, AsClient, PostgresStore, QueryError},
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply packages/graph/hash_graph/lib/graph/src/store/postgres/onthology/read.rs]: Applied in https://github.com/hashintel/hash/pull/875/commits/80991f8dac61cc0e018ae14d7fb0464eb2d22145

**Review by thehabbos007 — APPROVED** [review]
Looks good to me :tada:

**Pull request merged**